puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foremanPluginsRelease.groovy

```diff
@@ -0,0 +1,111 @@
+pipeline {
+    agent { label 'admin && sshkey' }
+
+    triggers {
+        cron('H H * * *')
+    }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Mash Koji Repositories') {
+            steps {
+
+                sh "ssh -o 'BatchMode yes' root@koji.katello.org foreman-mash-split-plugins.py"
+
+            }
+        }
+        stage('Repoclosure') {
+            steps {
+
+                parallel(
+                    'nightly/el7': { repoclosure('nightly', 'el7') },
+                    '1.16/el7': { repoclosure('1.16', 'el7') },
+                    '1.15/el7': { repoclosure('1.15', 'el7') },
+                    '1.15/f24': { repoclosure('1.15', 'f24') },
+                    '1.14/el7': { repoclosure('1.14', 'el7') },
+                    '1.14/f24': { repoclosure('1.14', 'f24') }
+                )
+
+            }
+        }
+        stage('Setup Push Environment') {
+            steps {
+
+                git url: 'https://github.com/theforeman/foreman-infra'
+                dir('deploy') { withRVM(["bundle install"]) }
+            }
+        }
+        stage('Push RPMs') {
+            steps {
+
+                parallel(
+                    'nightly': { push_rpms('nightly') },
+                    '1.16': { push_rpms('1.16') },
+                    '1.15': { push_rpms('1.15') },
+                    '1.14': { push_rpms('1.14') }
+                )
+
+            }
+            post {
+                always {
+                    deleteDir()
+                }
+            }
+        }
+    }
+}
+
+void push_rpms(version) {
+
+    dir('deploy') {
+
+        withRVM(["cap yum repo:sync -S overwrite=false -S merge=true -S repo_source=foreman-plugins-${version}/RHEL/7 -S repo_dest=plugins/${version}/el7"])
```

Comment:
Won't this skip f24 which 1.14 and 1.15 still have builds for?

Reply:
Yes - which I did intend to fix. However, from what I could tell, the current job has been not pushing F24 on those for quite some time (no excuse just pointing out holes I found).

Reply:
Looks like https://github.com/theforeman/foreman-infra/commit/11f911af51cc6449a830d7b771b0bbad5a0aaa68#diff-6f833b6369f41e3e65a2e93f31e495b4 dropped F24 indeed.

Reply:
Would this work if you add a ```dist``` parameter to ```push_rpms```?

```groovy
def source = 'RHEL/7'

if (dist == 'f24') {
    source = 'Fedora/24'
}

withRVM(["cap yum repo:sync -S overwrite=false -S merge=true -S repo_source=foreman-plugins-${version}/${source} -S repo_dest=plugins/${version}/${dist}"]) 
```